src/_nebari/stages/kubernetes_keycloak/__init__.py

```diff
@@ -315,6 +318,414 @@ def _attempt_keycloak_connection(
 
         print("Keycloak service successfully started")
 
+    def post_deploy(
+        self, stage_outputs: Dict[str, Dict[str, Any]], disable_prompt: bool = False
+    ):
+        """Restore Keycloak database (if backup exists) and create nebari-bot user after Keycloak is deployed."""
+
+        # Step 1: Restore database if backup exists
+        backup_file = Path(self.output_directory) / "keycloak-backup.sql"
+
+        if backup_file.exists():
+            print("\n" + "=" * 80)
+            print("KEYCLOAK DATABASE RESTORE")
+            print("=" * 80)
+            print(f"Found backup file: {backup_file}")
+            print(f"Size: {backup_file.stat().st_size / 1024:.2f} KB\n")
+
+            self._restore_keycloak_database(backup_file)
```

Comment:
What happens if the restore fails here? We aren't checking for that case.

Reply:
Added failure handling logic